packages/ods-react/src/components/combobox/src/context/useCombobox.tsx

```diff
@@ -0,0 +1,83 @@
+import type { ComboboxItemOrGroup } from '../components/combobox-item/ComboboxItem';
+import type { ComboboxValueChangeDetails } from '@ark-ui/react/combobox';
+import type { ComponentPropsWithRef } from 'react';
+import { type ReactNode, createContext, useContext } from 'react';
+
+export type ComboboxProp = Omit<ComponentPropsWithRef<'div'>, 'onSelect'> & {
+  allowCustomValue?: boolean,
+  children?: ReactNode,
+  className?: string,
+  customOptionRenderer?: (item: ComboboxItemOrGroup) => JSX.Element,
+  defaultValue?: string[] | undefined,
+  disabled?: boolean,
+  highlightResults?: boolean,
+  items: ComboboxItemOrGroup[],
+  newElementLabel?: string,
+  noResultLabel?: string,
+  onValueChange?: (value: ComboboxValueChangeDetails) => void,
+  readOnly?: boolean,
+  value?: string[]
+}
+
+interface ComboboxContextType {
+  allowCustomValue: boolean;
+  customOptionRenderer?: (item: ComboboxItemOrGroup) => ReactNode;
```

Comment:
`=> JSX.Element` to be inline with the `customOptionRenderer` return type of `ComboboxProp`